vcd/resource_vcd_vapp_vm_test.go

```diff
@@ -316,3 +317,126 @@ resource "vcd_vapp_vm" "{{.VmName2}}" {
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configText)
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckVcdVAppVmDestroy(vappName2),
+		Steps: []resource.TestStep{
+			{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm.test-vm", "name", "TestAccVcdVAppVm_SizingPolicy"),
+					resource.TestMatchResourceAttr(
+						"vcd_vapp_vm.test-vm", "sizing_policy_id", regexp.MustCompile("urn:vcloud:vdcComputePolicy:[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$")),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm.test-vm", "cpus", params["SizingPolicyCpus"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm.test-vm", "cpu_cores", params["SizingPolicyCpuCores"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_vm.test-vm", "memory", params["SizingPolicyMemory"].(string)),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdVAppVm_SizingPolicy = `
+resource "vcd_vm_sizing_policy" "test-sizing-policy" {
+  name        = "{{.Name}}"
+  description = "{{.Name}}"
+
+  cpu {
+    shares                = "886"
+    limit_in_mhz          = "3000"
+    count                 = "{{.SizingPolicyCpus}}"
+    speed_in_mhz          = "1000"
+    cores_per_socket      = "{{.SizingPolicyCpuCores}}"
+    reservation_guarantee = "0.55"
+  }
+
+  memory {
+    shares                = "885"
+    size_in_mb            = "{{.SizingPolicyMemory}}"
+    limit_in_mb           = "1024"
+    reservation_guarantee = "0.3"
+  }
+}
+
+resource "vcd_org_vdc" "test-vdc" {
+  name        = "{{.Name}}"
+  org         = "{{.Org}}"
+  description = "{{.Name}}"
+  default_vm_sizing_policy_id = vcd_vm_sizing_policy.test-sizing-policy.id
+  vm_sizing_policy_ids        = [vcd_vm_sizing_policy.test-sizing-policy.id]
+  
+  allocation_model = "Flex"
+  provider_vdc_name = "{{.ProviderVdc}}"
+  compute_capacity {
+    cpu {
+      allocated = 2048
+    }
+
+    memory {
+      allocated = 2048
+    }
+  }
+  storage_profile {
+    name    = "*"
+    limit   = 20240
+    default = true
+  }
+  enabled                  = true
+  enable_thin_provisioning = true
+  enable_fast_provisioning = true
+  delete_force             = true
+  delete_recursive         = true
+  elasticity = true
+  include_vm_memory_overhead = false
+}
+
+
+resource "vcd_vapp" "test-vapp" {
+  name = "{{.VappName}}"
+  org  = "{{.Org}}"
+  vdc  = vcd_org_vdc.test-vdc.name
+}
+
+resource "vcd_vapp_vm" "test-vm" {
+  org           = "{{.Org}}"
+  vdc           = vcd_org_vdc.test-vdc.name
+  vapp_name     = vcd_vapp.test-vapp.name
+  name          = "{{.Name}}"
+  computer_name = "foo"
+  catalog_name  = "{{.Catalog}}"
+  template_name = "{{.CatalogItem}}"
+  sizing_policy_id = vcd_vm_sizing_policy.test-sizing-policy.id
```

Comment:
Please add a comment explaining that the test is of NOT setting the CPU and memory - for fresh eyes it's invisible that this is what is tested. Even better, you could have those parameters but commented out + a comment explaining why.

Reply:
Done, in the new test cases (I've removed this file as we had already `resource_vcd_vapp_vm_with_vm_sizing_test.go` covering sizing policies)